telemetry/telemetry.proto

```diff
@@ -4,6 +4,57 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//Should be sent only after device resets
+message NameIDMessage { //Send the device ID with its corresponding name
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+}
+
+message StatisticalAggregate {
+  optional uint32 numSamples = 1;
+  optional uint32 min = 2;
+  optional uint32 max = 3;
+  optional uint32 mean = 4;
+  optional uint32 stdev = 5;
+}
+
+message GeneralReading { //this might want a more specific name?
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+	}
+	optional READING_TYPE type = 1;
```

Comment:
Perhaps READING_TYPE can be part of the header? It would mean that this isn't sent with every reading, and saves a byte or two every reading.

Additional metadata that might be relevant as part of the header:
- resolution (precision)
- accuracy
- units
- min / max (sensor) limits

(in Datalogger's protobuf definition, all this just gets stuffed into the arbitrary name string, but having this formalized could be useful)

Far future work potentially relevant metadata per reading:
- frequency distribution between samples, perhaps a min/max time between samples - useful for debugging, but perhaps not so much for race operations